plugins/modules/mso_schema_template_anp_epg_useg_attribute.py

```
@@ -0,0 +1,258 @@
+    template = module.params.get("template").replace(" ", "")
+    anp = module.params.get("anp")
+    epg = module.params.get("epg")
+    name = module.params.get("name")
+    description = module.params.get("description")
+    attribute_type = module.params.get("type")
+    value = module.params.get("value")
+    operator = module.params.get("operator")
+    useg_subnet = module.params.get("useg_subnet")
+    state = module.params.get("state")
+    mso = MSOModule(module)
+
+    if state == "present":
+        if attribute_type in ["mac", "dns"] and value is None:
+            mso.fail_json(msg="Failed due to invalid 'value' and the attribute_type is: {0}.".format(attribute_type))
+        elif attribute_type not in ["mac", "dns", "ip"] and (value is None or operator is None):
+            mso.fail_json(msg="Failed due to invalid 'value' or 'operator' and the attribute_type is: {0}.".format(attribute_type))
+
+    mso_schema = MSOSchema(mso, schema, template)
+    mso_schema.set_template(template)
+    mso_schema.set_template_anp(anp)
+    mso_schema.set_template_anp_epg(epg)
+
+    if mso_schema.schema_objects["template_anp_epg"].details.get("uSegEpg"):
+        mso_schema.set_template_anp_epg_useg_attr(name, fail_module=False)
+        if mso_schema.schema_objects["template_anp_epg_useg_attribute"] is not None:
+            useg_attr_path = "/templates/{0}/anps/{1}/epgs/{2}/uSegAttrs/{3}".format(
+                template, anp, epg, mso_schema.schema_objects["template_anp_epg_useg_attribute"].index
+            )
+            mso.existing = mso_schema.schema_objects["template_anp_epg_useg_attribute"].details
+    else:
+        mso.fail_json(msg="{0}: is not a valid uSeg EPG.".format(epg))
+
+    if state == "query":
+        if name is None:
+            mso.existing = mso_schema.schema_objects["template_anp_epg"].details.get("uSegAttrs")
+        elif not mso.existing:
+            mso.fail_json(msg="The uSeg Attribute: {0} not found.".format(name))
```

It would be good to check if we should fail the module if an object is not found on query here as well..

---

Yes, we fail the module when we query a specific object.